src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -716,9 +722,13 @@ internal static class CoreTypes
                     { typeof(void),                                        new[] { "void" } },
                     { typeof(IPAddress),                                   new[] { "ipaddress" } },
                     { typeof(DscLocalConfigurationManagerAttribute),       new[] {"DscLocalConfigurationManager"}},
+                    { typeof(WildcardPattern),                             new[] { "WildcardPattern" } },
+                    { typeof(X509Certificate),                             new[] { "X509Certificate" } },
+                    { typeof(X500DistinguishedName),                       new[] { "X500DistinguishedName" } },
                     { typeof(XmlDocument),                                 new[] { "xml" } },
+                    { typeof(CimSession),                                  new[] { "CimSession" } },
```

Comment:
>// Do not add types to this pool unless they can be safely exposed to an attacker and will not
// expose the ability to corrupt or escape PowerShell's environment. The following operations must
// be safe: type conversion, all constructors, all methods (instance and static), and
// and properties (instance and static).

Have you done a security review on those newly added types?